### PR TITLE
base vitest setup

### PR DESCRIPTION
**Pull request by jeetiss**
- setup `vitest`
- move tests to `typescript`
- ~add a lot of `// @ts-expect-error` comments~

### CODE REVIEW

File: packages/wouter-preact/package.json [hunk 1]
```diff
@@ -35,12 +35,11 @@
       "default": "./esm/static-location.js"
     }
   },
-  "module": "esm/index.js",
-  "types": "types/ts3.9.4/index.d.ts",
+  "types": "types/index.d.ts",
   "typesVersions": {
```
Comment (molefrog) [inline packages/wouter-preact/package.json:39]: Could you check what versions of TS support type declaration paths defined in `exports`? I've got a feeling that we don't need `typesVersions` at all now.

File: packages/wouter-preact/test/preact.test.tsx [hunk 2]
```diff
@@ -33,15 +33,15 @@ describe("Preact support", () => {
 
         <main data-testid="routes">
           <Switch>
-            Welcome to the list of {100} greatest albums of all time!
+            <>Welcome to the list of {100} greatest albums of all time!</>
             <Route path="/albums/all">Rolling Stones Best 100 Albums</Route>
-            <Route path="/albums/:name">
+            <Route<{ name: string }> path="/albums/:name">
```
Comment (molefrog) [inline packages/wouter-preact/test/preact.test.tsx:38]: Hmm, routes should infer the proper type of parameters from the path.

File: packages/wouter-preact/package.json (hunk 1 above)
Comment (jeetiss) [reply packages/wouter-preact/package.json:39]: nope, `type: module` doesn't affect how typescript resolves types. `typesVersions` is used when `moduleResolution` is set to `node`, and the `type` field in `exports` is used when `moduleResolution` is `nodenext` or `node16` or `bundler`

Comment (molefrog) [reply packages/wouter-preact/package.json:39]: ok!

**Review by molefrog — APPROVED** [review]
Looks good! Could you pls check why route parameters are not inferred?

File: packages/wouter/test/use-location.test.tsx [hunk 3]
```diff
@@ -31,16 +32,17 @@ describe("`value` first argument", () => {
     unmount();
   });
 
-  it("supports history.back() navigation", () => {
-    jest.useFakeTimers();
+  // Vitest doesn't support Jest's legacy timers.
+  it.skip("supports history.back() navigation", () => {
```
Comment (molefrog) [inline packages/wouter/test/use-location.test.tsx:36]: I don’t even remember why I had to use fake timers here. Does the test pass when you remove these calls?

File: packages/wouter-preact/test/preact.test.tsx (hunk 2 above)
Comment (jeetiss) [reply packages/wouter-preact/test/preact.test.tsx]: I managed to fix this

File: packages/wouter/test/use-location.test.tsx (hunk 3 above)
Comment (jeetiss) [reply packages/wouter/test/use-location.test.tsx]: I replaced `FakeTimers` with `waitFor` util from `testing-library`

**Comment by molefrog** [comment]
Nice, thank you!

**Pull request merged**